spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+var languageNegotiationLambda = require( '../../src/language_negotiation_lambda' );
+
+describe( 'languageNegotiationLambda', function() {
+    var event,
+        default_header = ['en'];
+
+    beforeEach(function(){
+        // Cloudfront Request Object
+        event = {
+            "Records": [
+                {
+                  "cf": {
+                    "request": {
+                      "headers": {
+                        "Host": [
+                          "localhost"
+                        ],
+                        "User-Agent": [
+                          "Test Agent"
+                        ]
+                      },
+                      "clientIp": "2001:cdba::3257:9652",
+                      "httpVersion": "2.0",
+                      "uri": "/",
+                      "method": "GET"
+                    },
+                    "config": {
+                      "distributionId": "EXAMPLE"
+                    }
+                  }
+                }
+            ]
+        };
+    });
+
+    describe( 'Valid cookies', function(){
+        [
+            { cookie: "en",     expected: "en" },
+            { cookie: "EN",     expected: "en" },
+            { cookie: "es",     expected: "es" },
+            { cookie: "ES",     expected: "es" },
+            { cookie: "en-US",  expected: "en" },
+            { cookie: "es-419", expected: "es" },
+            { cookie: "es_419", expected: "es" },
+            { cookie: "fr",     expected: "es" },
+            { cookie: "",       expected: "es" }
```

Comment:
This seems to be mixing concerns, as it is testing both that the cookie logic is able to resolve varying forms of valid cookies to `es`, and that it falls back to `Accept-Language` when the cookie resolves to something unsupported. I think it would be clearer if we tested those cases separately.